### PR TITLE
Feature/test coverage improval

### LINKED ISSUE #1: Rewrite test_version
Right now test_version is a dummy test that does more harm than it's helpful. I want to rewrite the test so it compares the version in __init__, that defines the pystatis.__version__ attribute, with the version stated in the pyproject.toml file. This would assure that we'll never ever forget to updated both again (should even be a pre commit hook)

### LINKED ISSUE #24: Add Test Coverage and "Documentation Building" Badge to README

### PR DESCRIPTION
**Pull request by MarcoHuebner**
Pull request contains suggestion/ solution for #1 (and reintroduces the version test for compatibility with the version specified in the sphinx doc), increases test coverage to >95%, and is compatible with previous PR.

resolves #1 , resolves #24

### CODE REVIEW

File: .github/workflows/run-tests.yaml
Comment (pmayd) [inline .github/workflows/run-tests.yaml:22]: sollten wir vielleicht aktivieren?

File: .github/workflows/run-tests.yaml
Comment (pmayd) [inline .github/workflows/run-tests.yaml:51]: use `${{matrix.poetry-version}}`?

File: .github/workflows/run-tests.yaml
Comment (pmayd) [inline .github/workflows/run-tests.yaml:50]: Use `${{matrix-python-version}}`?

File: README.md
Comment (pmayd) [inline README.md:134]: Is it not possible to link to the created artifact as latest cicd run, pointing to html files which should be rendered? At least that is how I did it in another project but with Atlassian Bamboo and not Github but I think the artifact should be available as resource? Is this artifact a folder containing html or always an archive? Can we configure this?

File: docs/source/conf.py
Comment (pmayd) [inline docs/source/conf.py:1]: Is this file generated by hand or autogenerated from the README/pyproject.toml?

File: docs/source/contribute.rst
Comment (pmayd) [inline docs/source/contribute.rst:21]: We no longer recommend doing it via `pipx`, this has some problems I ran into and you should now always install a fresh poetry in the current venv, see README, maybe just copy over the latest steps from the most recent README

File: src/pystatis/__init__.py
Comment (pmayd) [inline src/pystatis/__init__.py:20]: nice

File: tests/rsc/data.zip
Comment (pmayd) [inline tests/rsc/data.zip:1]: Should we have a new zip file here?

File: tests/test_table.py
Comment (pmayd) [inline tests/test_table.py:1]: wow did you really add all these tests? When did you find the time for this ^^

**Review by pmayd — APPROVED** [review]
LGTM

I see now that you are using the new test data in the new test files so good by me! I did not test it locally but I guess we can talk more about it once we have merged it and see what is still missing. Would be nice to understand which `.rst` files are autogenerated and which ones have to be maintained manually, though!

File: .github/workflows/run-tests.yaml
Comment (MarcoHuebner) [reply .github/workflows/run-tests.yaml:22]: Finde es eigentlich gut, dann kann man alle Fehler auf einmal abarbeiten und geht nicht push für push die neu auftauchenden Fehler durch 😄

File: .github/workflows/run-tests.yaml
Comment (MarcoHuebner) [reply .github/workflows/run-tests.yaml:37]: Im neusten dev-branch hotfix ist es jetzt global definiert, mMn noch bessere/ konsistentere Lösung - lass mich wissen was du davon hältst

File: .github/workflows/run-tests.yaml
Comment (MarcoHuebner) [reply .github/workflows/run-tests.yaml:51]: Für nächstes Meeting: DIskussion ob matrix - falls wir mehrere poetry Versionen unterstützen wollen - oder "nur" globale Variable

File: docs/source/conf.py
Comment (MarcoHuebner) [reply docs/source/conf.py:1]: Should be hand-crafted afaik

File: .github/workflows/run-tests.yaml
Comment (pmayd) [reply .github/workflows/run-tests.yaml:22]: Das macht man aber lokal und nicht remote :) aber ja ist hier nicht relevant weil Test ein einziger step ist und der sowieso alle Fehler anzeigt, geht eher um längere cicd pipelines, die dann nicht sinnlos weiterlaufen sollen

File: .github/workflows/run-tests.yaml
Comment (pmayd) [reply .github/workflows/run-tests.yaml:51]: nein für poetry brauchen wir das nicht, das ist nur ein Hilfstool und da ist mir nicht bekannt, dass man sowas braucht. Globale Variable ist hier am besten, hatte ich auch im Meeting gemeint, gut, dass du das gefunden hast

File: .github/workflows/run-tests.yaml
Comment (pmayd) [reply .github/workflows/run-tests.yaml:37]: perfekt

**Comment by MarcoHuebner** [comment]
The branch is now up to date with the current dev. I removed all Sphinx-related updates (besides one dummy test).

> Would be nice to understand which `.rst` files are autogenerated and which ones have to be maintained manually, though!

Let me know if you already have ideas on notes regarding updating the basic `.rst` files - mention this in the `Contributor information` section of the `Readme`?

**Comment by MarcoHuebner** [comment]
TODO: Double-check config tests, since they seem to mess with the local config in undesired ways - DO NOT MERGE YET

**Comment by pmayd** [comment]
That is unlikely as I tested this several times and my local config was never changed but maybe You can describe your observations. It is a little tricky to test this but the tests should use a temporary folder and config and not your local one

**Comment by MarcoHuebner** [comment]
The `config_dir` in the `settings.ini` was/ is pointing to a local, temporary pytest directory (`...\AppData\Local\Temp\pytest-of-...\pytest-222\.pystatis18`) after running `pytest`. I haven't investigated the behavior in detail yet, but I suppose it results from the `config_dir` fixture and one of the `create_settings` calls incorrectly overwriting the "real" `settings.ini`.

Is the settings still used in the when #20 is addressed? Otherwise, this could be ignored.

**Comment by pmayd** [comment]
> The `config_dir` in the `settings.ini` was/ is pointing to a local, temporary pytest directory (`...\AppData\Local\Temp\pytest-of-...\pytest-222\.pystatis18`) after running `pytest`. I haven't investigated the behavior in detail yet, but I suppose it results from the `config_dir` fixture and one of the `create_settings` calls incorrectly overwriting the "real" `settings.ini`.
> 
> Is the settings still used in the when #20 is addressed? Otherwise, this could be ignored.

We don't use a settings.ini any more, everything is directly stored in the config.ini 

But I just realize I might have made an error and we need it.... Once you finish your session the information is lost where to find the new fine that is stupid from my side so I actually need the settings.ini again ...

Maybe we can just omit the feature that you can configure your own config dir

File: README.md
Comment (MarcoHuebner) [reply README.md]: Linking to the Actions / Workflow would be possible but unspecific, linking to a specific artifact will get outdated quickly - would thus advocate for not linking it somewhere since I'm not aware of good and persistent solution

Comment (pmayd) [reply README.md]: Of course, I meant linking the latest, normally cicd provides the possibility to link the latest successful run, I will have a look

Comment (pmayd) [reply README.md]: But doesn't matter as we decided to do the github pages variant, so please continue!

**Comment by MarcoHuebner** [comment]
Continued from internal branch in #45

**Pull request closed without merging**
